app/src/main/java/unipd/se18/ocrcamera/MainActivity.java

```diff
@@ -0,0 +1,123 @@
+asked for accessing the camera. --> Check the official Android Developer tutorials for the guide I used
+to manage permissions (https://developer.android.com/guide/topics/media/camera).
+ */
+
+public class MainActivity extends AppCompatActivity {
+
+    private Button takePictureButton;
+    private ImageView imageView;
+    private Uri file;
+    private TextView outText;
+    private int cameraCode = 100;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        takePictureButton = (Button) findViewById(R.id.button);
+        imageView = (ImageView) findViewById(R.id.pictureShot);
+        outText = (TextView) findViewById(R.id.outText);
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
+            takePictureButton.setEnabled(false);
+            ActivityCompat.requestPermissions(this, new String[] {
+                    Manifest.permission.CAMERA, Manifest.permission.WRITE_EXTERNAL_STORAGE }, 0);
+        }
+    }
+
+    /**
+    This method set the button enabled if the permmissions asked are granted by the user
+    @param requestCode int camera enabler code when permissions granted
+    @param permissions String[] contains result of checking if permissions are declared in manifest file
+    @author Andrea Ton
+     */
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+        if (requestCode == 0) {
+            if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED
+                    && grantResults[1] == PackageManager.PERMISSION_GRANTED) {
+                takePictureButton.setEnabled(true);
+            }
+        }
+    }
+
+    /**
+    This method implements also the OCR using the wrapper for the mlkit api,
+    the OCR class is not asked for review.
+    @param view View layout object that call this method (in this case a button)
+    @author Andrea Ton
+     */
+    public void takePicture(View view) {
+        Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        file = Uri.fromFile(getOutputMediaFile()); //creates a uri from a file
+        intent.putExtra(MediaStore.EXTRA_OUTPUT, file);
+        if (intent.resolveActivity(getPackageManager()) != null) {
+            startActivityForResult(intent, cameraCode);
+            WrapperMLKit scanner = new WrapperMLKit(getApplicationContext(),file);
+            String ingredients = scanner.getTextFromImg(BitmapFactory.decodeFile(file.getPath()));
+            outText.setText(ingredients);
+        }
+    }
+
+    /**
+    @param requestCode int arbitrary constant used to identify activity requested
+    @param resultCode int system code for activity result
+    @param data Intent returned from activity called
+    @author Andrea Ton
+     */
```

Comment:
describe what you want this method to do in the specifications as you have done with the others